submissions/cynthiaNwakaeme/domManipulation/signUp.js

```diff
@@ -0,0 +1,103 @@
+let submitForm = document.getElementById("validateForm"),
+    firstName = document.getElementById("firstName"),
+    lastName = document.getElementById("lastName"),
+    phoneNumber = document.getElementById("phoneNumber"),
+    email = document.getElementById("email"),
+    password = document.getElementById("password"),
+    confirmPassword = document.getElementById("confirmPassword");
+
+// 2. Name Field Validation (To ensure input name has only alphabet)
+// nameRegex: Any character from upper A to lower z
+function validate(field, event, result, output) {
+    const nameRegex = /^[a-zA-Z]/;
+
+    if (field.value === "") {
+        field.nextElementSibling.innerHTML = result; //When field is empty
+        event.preventDefault();
+        return false;
+    } else if (!field.value.match(nameRegex)) {
+        field.nextElementSibling.innerHTML = output; //When name has special characters or numbers
+        event.preventDefault();
+        return false;
+    } else {
+        field.nextElementSibling.innerHTML = "";
+        return true;
+    }
+}
+// 2. Phone Field Validation (Ensure only integers and + are allowed)
+//phoneRegex: I considered international numbers and office lines( >= 8 digits )
+function phoneValidation(field, event) {
+    const phoneRegex = /^[\+]?[0-9]{8,14}$/i;
+
+    if (field.value === "") {
+        field.nextElementSibling.innerHTML = "Please enter your Phone Number"; //When field is empty
+        event.preventDefault();
+        return false;
+    } else if (!field.value.match(phoneRegex)) {
+        field.nextElementSibling.innerHTML = "Please enter a valid Phone Number"; //When field has not a mail format
+        event.preventDefault();
+        return false;
+    } else {
+        field.nextElementSibling.innerHTML = "";
+        return true;
+    }
+}
+
+// 3. Email Field Validation (To ensure input type is in an email-address format)
+// emailRegex: email format (outlook.com, gmail.com etc)
+function emailValidation(field, event) {
```

Comment:
Well done 

These 3 function `validate`, `phoneValidation` and `emailValidation` are the same. The only difference is their `field`, `regex` and `error message`. I will like for you to handle this within a single function. Then pass the rest as a parameter

Reply:
Hello @adaugochi Thank you so much. I have worked on it Kindly approve my PR

Reply:
Hello @adaugochi . I really do appreciate the fact you took your time to explain this to me.
I have effected the changes ma.


Kindly approve my PR.
Thank You so much.